Add documentation for new /google-pay endpoints (#624)
Basis-Theory/developers.basistheory.com: docs/api/google-pay/api.mdx

---
title: Google Pay™
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

import { HttpMethod, HttpMethods } from "@site/src/components/shared/HttpMethod";
import { Permission } from "@site/src/components/shared/Permission";
import { Alert, Alerts } from "@site/src/components/shared/Alert";

# Google Pay™

## Create

> Decrypts a Google Pay token and creates either a Google Pay token or a Basis Theory's [Token Intent](/docs/api/tokens/token-intents#token-intent-object) based on the provided ecrypted token. This endpoint only supports the `ECv2` token versions. It supports both auth methods, `PAN_ONLY` and `CRYPTOGRAM_3DS`.

Reviewer: ```suggestion
Decrypts a Google Pay token and creates either a Google Pay token or a Basis Theory's [Token Intent](/docs/api/tokens/token-intents#token-intent-object) based on the provided encrypted token. This endpoint only supports the `ECv2` token versions. It supports both auth methods, `PAN_ONLY` and `CRYPTOGRAM_3DS`.
```

| Auth Method   | Token Type   |
|---------------|--------------|
| [PAN_ONLY](https://developers.google.com/pay/api/web/guides/resources/payment-data-cryptography#pan_only) | [`card`](/docs/api/tokens/token-types#card) |
| [CRYPTOGRAM_3DS](https://developers.google.com/pay/api/web/guides/resources/payment-data-cryptography#cryptogram_3ds)  | [`network_token`](/docs/api/tokens/token-types#network-token)  |

<HttpMethod method={HttpMethods.POST} endpoint="https://api.basistheory.com/google-pay" />

#### Permissions

<Permission content="google-pay:create" />

### Request

<Tabs className="bt-tabs" groupId="languages">
  <TabItem value="shell" label="cURL">

    ```shell showLineNumbers
    curl --request POST \
    --url https://api.basistheory.com/google-pay \
    --header 'BT-API-KEY: <API_KEY>' \
    --header 'Content-Type: application/json' \
    --data '{
      "expires_at": "2030-12-25"
      "google_payment_data": {
        "protocolVersion": "...",
        "signature": "...",
        "intermediateSigningKey": {
          "signedKey": "...",
          "signatures": ["...", "..."]
        },
        "signedMessage": "...
      }
    }'
    ```
  </TabItem>

>   <TabItem value="node" label="Node">

Reviewer: we'll want to generate those SDKs before we merge!

    ```javascript
    await client.googlePay.create();
    ```
  </TabItem>
  <TabItem value="csharp" label="C#">
    ```csharp showLineNumbers
    await client.GooglePay.CreateAsync(new GooglePayCreateRequest());
    ```
  </TabItem>
  <TabItem value="python" label="Python">
    ```python showLineNumbers
    client.google_pay.create()
    ```
  </TabItem>
</Tabs>

#### Request Parameters

| Attribute  | Required | Description                                                        |
|------------|----------|--------------------------------------------------------------------|
| `expires_at` | false | An optional expiration date for the token. If blank, no expiration date is set for the token. |
| `google_payment_data` | true | The Google Pay [token](https://developers.google.com/pay/api/web/guides/resources/payment-data-cryptography#payment-method-token-structure). |

### Response

Returns a Google Pay Token for tokenized cards that consists of device PAN and cryptograms.

> The return will include the Card BIN enrichments according to the tenant settings.

Reviewer: ```suggestion
```
I realized this is wrong on the Apple Pay docs too 😬 we always return `card` info regardless of Tenant settings

Returns [an error](/docs/api/errors) if there were validation errors, if the Google Pay token failed to decrypt or the token failed to create.

```json
{
  "google_pay": {
    "id": "b7248b11-acfb-4695-93a9-9609f90ec038",
    "tenant_id": "...",
    "status": "active",
    "expires_at": "2030-12-25T00:00:00+00:00",
    "created_by": "50c63e9b-91e5-4a6a-9f9e-81088d106f2c",
    "created_at": "2025-08-06T11:02:53.9261435+00:00",
    "card": {
      "bin": "411111",
      "last4": "1111",
      "expiration_month": 12,
      "expiration_year": 2027,
      "brand": "visa",
      "funding": "debit",
      "authentication": "sca_required"
    },
    "details": {
        "tsp": "Google",
        "auth_method": "PAN_ONLY",
        "message_id": "AH2EjtdkuuxxO3nA7RuWAo0fU4u5BPoySvd9ajlSSrTWy3yTCu2IRbRwp8K3sFjSUS-ELa7UzetwY6UFsNLISYiKY6arNIOfYxutYYH3LJdIH0UcmoRDzmM",
        "eci_indicator": "",
        "assurance_details": {
          "account_verified": true,

>           "card_holder_authenticated": false

Reviewer: small indentation problem with these lines

        }
    }
  }
}
```

<Alert type={Alerts.Information}>

>   For encrypted cards that consist of PAN a Token Intent will be created. Refer to Basis Theory's [Token Intents documentation](/docs/api/tokens/token-intents#token-intent-object) for more information.

Reviewer: ```suggestion
  For encrypted cards with an authorization method of "PAN_ONLY", a Token Intent will be created. Refer to [Token Intents documentation](/docs/api/tokens/token-intents#token-intent-object) for more information.
```

</Alert>

## Retrieve

> Retrieves an already tokenized Google Pay token. For encrypted cards that consist of PAN look at Basis Theory's [Token Intents documentation](/docs/api/tokens/token-intents#get-a-token-intent).

Reviewer: ```suggestion
Retrieves an already tokenized Google Pay token. For encrypted cards with an authorization method of "PAN_ONLY" use [Token Intents documentation](/docs/api/tokens/token-intents#get-a-token-intent).
```

  },
  "authentication": {
    "threeds_cryptogram": "AAAAAA=",
    "eci_indicator": "7"
  }
}
```

Google Pay Tokens created from a Google Pay token will contain a `details` attribute.

| Attribute                         | Type      | Description                                   |
|-----------------------------------|-----------|-----------------------------------------------|
| `details`                         | Object    | Details related to the transaction service provider (TSP).  |
| `tsp`                             | String    | The name of the transaction service provider. |
| `auth_method`                     | String    | The authentication method used.              |
| `message_id`                      | String    | Unique identifier for the message.           |
| `eci_indicator`                   | String    | Electronic Commerce Indicator (may be empty).|
| `assurance_details`               | Object    | Information about the assurance of the transaction. |
| `account_verified`                | Boolean   | Indicates if the account was verified.       |
| `card_holder_authenticated`       | Boolean   | Indicates if the cardholder was authenticated.|

## Delete

> Deletes an already tokenized Google Pay token. For encrypted cards that consist of PAN look at Basis Theory's [Token Intents documentation](/docs/api/tokens/token-intents#delete-token-intent).

Reviewer: ```suggestion
Deletes an already tokenized Google Pay token. For encrypted cards with an authorization method of "PAN_ONLY" use [Token Intents documentation](/docs/api/tokens/token-intents#delete-token-intent).
```
